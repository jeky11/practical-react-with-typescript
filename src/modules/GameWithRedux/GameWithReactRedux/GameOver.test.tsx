import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useSelector, useDispatch } from 'react-redux';

import { store } from '@/store';

import { GameOver } from './GameOver';

jest.mock('react-redux', () => ({
  __esModule: true,
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

describe('GameOver test cases', () => {
  it('GameOver with loose check', () => {
    const mockDispatch = jest.fn();

    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
    (useSelector as jest.Mock).mockReturnValue({
      isGameOver: true,
      isWin: false,
    });

    const { asFragment } = render(
      <Provider store={store}>
        <GameOver />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();

    userEvent.click(screen.getByText('🙁'));

    expect(mockDispatch).toHaveBeenCalled();
  });
  it('GameOver with win check', () => {
    const mockDispatch = jest.fn();

    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
    (useSelector as jest.Mock).mockReturnValue({
      isGameOver: true,
      isWin: true,
    });

    const { asFragment } = render(
      <Provider store={store}>
        <GameOver />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();

    userEvent.click(screen.getByText('😎'));

    expect(mockDispatch).toHaveBeenCalled();
  });
});
