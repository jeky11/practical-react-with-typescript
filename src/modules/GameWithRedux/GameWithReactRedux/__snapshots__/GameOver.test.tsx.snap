// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOver test cases GameOver with loose check 1`] = `
<DocumentFragment>
  .emotion-0 {
  top: 60%;
  left: 50%;
  z-index: 11;
  width: 8vw;
  height: 8vw;
  font-size: 4vw;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  line-height: 8vw;
  position: absolute;
  border-radius: 50%;
  pointer-events: auto;
  background-color: #d1d1d1;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  border: 1px solid rgba(51, 51, 51, 0.25);
}

<div
    class="emotion-0"
  >
    üôÅ
  </div>
</DocumentFragment>
`;

exports[`GameOver test cases GameOver with win check 1`] = `
<DocumentFragment>
  .emotion-0 {
  top: 60%;
  left: 50%;
  z-index: 11;
  width: 8vw;
  height: 8vw;
  font-size: 4vw;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  line-height: 8vw;
  position: absolute;
  border-radius: 50%;
  pointer-events: auto;
  background-color: #d1d1d1;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  border: 1px solid rgba(51, 51, 51, 0.25);
}

<div
    class="emotion-0"
  >
    üòé
  </div>
</DocumentFragment>
`;
